src/lightning/fabric/utilities/distributed.py

```diff
@@ -23,9 +25,62 @@ class group:  # type: ignore
         WORLD = None
 
 
+if TYPE_CHECKING:
+    from lightning.fabric import Fabric
+    from lightning.fabric.strategies import Strategy
+
+
 log = logging.getLogger(__name__)
 
 
+def is_shared_filesystem(fabric: "Fabric", path: Optional[_PATH] = None, timeout: int = 3) -> bool:
+    """Checks whether the filesystem under the given path is shared across all processes.
+
+    Args:
+        fabric: The current :class:`~lightning.fabric.fabric.Fabric` instance
+        path: The path to check. Defaults to the current working directory. The user must have permissions to write
+            to this path or the parent folder, and the filesystem must be writable.
```

Comment:
When would you want to pass a path? Could we use /tmp/ as a default for this?

Reply:
The path would typically be where you save or load the data. You might have different locations mounted on your system, where e.g. /data and /home is shared across the cluster but /tmp is not. The path is kinda the most important thing in this fuction, because otherwise we wouldn't know where to look.